satpy/tests/reader_tests/test_viirs_fires.py

```diff
@@ -0,0 +1,201 @@
+
+DEFAULT_M13_FILE_DTYPE = np.float32
+DEFAULT_M13_FILE_DATA = np.arange(start=300, stop=340, step=0.4,
+                                  dtype=DEFAULT_M13_FILE_DTYPE).reshape(DEFAULT_FILE_SHAPE)
+
+DEFAULT_POWER_FILE_DTYPE = np.float32
+DEFAULT_POWER_FILE_DATA = np.arange(start=1, stop=25, step=0.24,
+                                    dtype=DEFAULT_POWER_FILE_DTYPE).reshape(DEFAULT_FILE_SHAPE)
+
+
+class FakeFiresNetCDF4FileHandler(FakeNetCDF4FileHandler):
+    """Swap in CDF4 file handler"""
+    def get_test_content(self, filename, filename_info, filename_type):
+        """Mimic reader input file content"""
+        file_content = {}
+        file_content['/attr/satellite_name'] = "AFEDR_npp"
+        file_content['/attr/instrument_name'] = 'VIIRS'
+
+        file_content['Fire Pixels/FP_latitude'] = DEFAULT_LATLON_FILE_DATA
+        file_content['Fire Pixels/FP_longitude'] = DEFAULT_LATLON_FILE_DATA
+        file_content['Fire Pixels/FP_power'] = DEFAULT_POWER_FILE_DATA
+        file_content['Fire Pixels/FP_T13'] = DEFAULT_M13_FILE_DATA
+        file_content['Fire Pixels/FP_confidence'] = DEFAULT_DETECTION_FILE_DATA
+        file_content['Fire Pixels/attr/units'] = 'none'
+        file_content['Fire Pixels/shape'] = DEFAULT_FILE_SHAPE
+
+        # convert to xarrays
+        from xarray import DataArray
+        for key, val in file_content.items():
+            if isinstance(val, np.ndarray):
+                attrs = {}
+                for a in ['FP_latitude', 'FP_longitude',  'FP_T13', 'FP_confidence']:
+                    if key + '/attr/' + a in file_content:
+                        attrs[a] = file_content[key + '/attr/' + a]
+                if val.ndim > 1:
+                    file_content[key] = DataArray(val, dims=('fakeDim0', 'fakeDim1'), attrs=attrs)
+                else:
+                    file_content[key] = DataArray(val, attrs=attrs)
+
+        return file_content
+
+
+class FakeFiresTextFileHandler(BaseFileHandler):
+    def __init__(self, filename, filename_info, filetype_info, **kwargs):
+        """Get fake file content from 'get_test_content'"""
+        super().__init__(filename, filename_info, filetype_info)
+        self.file_content = self.get_test_content()
+
+    def get_test_content(self):
+        fake_file = io.StringIO('''\n\n\n\n\n\n\n\n\n\n\n\n\n\n
+        24.64015007, -107.57017517,  317.38290405,   0.75,   0.75,   40,    4.28618050
+        25.90660477, -100.06127167,  331.17962646,   0.75,   0.75,   81,   20.61096764''')
+
+        return dd.from_pandas(pd.read_csv(fake_file, skiprows=15, header=None,
+                                          names=["latitude", "longitude",
+                                                 "T13", "Along-scan", "Along-track",
+                                                 "detection_confidence",
+                                                 "power"]), chunksize=1)
+
+
+class TestVIIRSACTIVEFIRES_CDF4(unittest.TestCase):
```

Comment:
Please make these PEP8 names: `TestVIIRSActiveFiresNetCDF4` or something like that. `VIIRS` is an acronym so it should be all caps. Other class name needs to be updated too.